pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -805,8 +805,18 @@ stdenvNoCC.mkDerivation {
     # number of general purpose registers. Keeping EBP available provides
     # non-trivial performance benefits.
     + optionalString (!targetPlatform.isx86_32) ''
-      echo " -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer " >> $out/nix-support/cc-cflags-before
+      echo " -fno-omit-frame-pointer " >> $out/nix-support/cc-cflags-before
     ''
+    +
+      optionalString
+        (
+          targetPlatform.isx86_64
+          || targetPlatform.isAarch64
+          || (targetPlatform.isRiscV && versionAtLeast ccVersion "15.1")
```

Comment:
isRiscV means riscv32 and riscv64, is mno-omit-leaf-frame-pointer supported for riscv32?
Also what if cc is clang?

Reply:
It refers to both 32-bit and 64-bit. Clang should handle it just fine. Though, idk if RV32 supports this but there's a good chance. I'd say just spin up a cross build and see if the compiler takes it.

Reply:
Afaict, it's been around long enough that we don't need a version check for clang.

Reply:
Then the check needs to differentiate between gcc and clang